### ISSUE TITLE
crash trying to report UIAlertController event

### ISSUE THREAD

**Issue by Busta117** [segmentio/analytics-ios#729]
app crash when `recordScreenViews` is true and try to record a screen visit automatically of an UIAlertController... 
crash happens because UIAlertController doesn't have any title, so this line crash (line 243 of SEGAnalytics.m)

```objective-c
NSCAssert1(screenTitle.length > 0, @"screen name (%@) must not be empty.", screenTitle);
```

**Comment by f2prateek** [comment]
Automating screen tracking requires all screens to have titles.
We could relax the check though to warn instead of crash. The downside of that would be that most people often miss details in the logs and could be confused why certain screens don't get tracked.

**Comment by f2prateek** [comment]
(sorry, accidentally closed!)

**Comment by Busta117** [comment]
yeah, have sense, but i saw the problem, maybe only happens in swift, when you try to get the title, and view doesn't have any title Segment try to get the name of the class and remove the `ViewController` text and create the report with that, but that validation is over a nil instead of empty string and that `[top title]` returns and empty string

i'm talking about `UIViewController+SEGScreen.m` file, line 69

i'm uploading a PR with the fix

**Comment by f2prateek** [comment]
Closed by #730
